### PR TITLE
restrict some global JavaScript objects

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

* Removed server-side JavaScript object `ArangoClusterComm`, so it cannot be used from inside JavaScript operations or Foxx. The `ArangoClusterComm` object was previously used inside a few internal JavaScript operations, but was not part of the public APIs.

* Restrict access to functions inside JavaScript objects `ArangoAgency` and `ArangoAgent` to JavaScript code that is running in privileged mode, i.e. via the server's emergency console, the `/_admin/execute` API (if turned on) or internal bootstrap scripts.

* Added startup option `--javascript.transactions` to allow turning off JavaScript transactions if not needed. The default value for this option is `true`, meaning JavaScript transactions are available as before. However, using this option they can be turned off by admins to limit the amount of JavaScript user code that is executed.

It is still unclear to what extend this should or can be backported.

This still lacks a few more tests, which will be added soon.

- [x] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

- [x] Docs PR: TO DO

### Testing & Verification

- [x] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as *all*.
- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [x] Added new **integration tests** (e.g. in server_permissions)

Link to Jenkins PR run:
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13489/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests blue

File: js/common/modules/@arangodb/graph/traversal.js [hunk 1]
```diff
@@ -105,7 +104,7 @@ function collectionDatasourceFactory (edgeCollection) {
 
   // we can call the "fast" version of some edge functions if we are
   // running server-side and are not a coordinator
-  var useBuiltIn = (typeof ArangoClusterComm === 'object');
+  var useBuiltIn = require('internal').isArangod();
```
Comment (mchacki) [inline js/common/modules/@arangodb/graph/traversal.js:107]: THis is a much better test :+1:

File: js/apps/system/_admin/aardvark/APP/statistics.js [hunk 2]
```diff
@@ -467,94 +464,10 @@ router.get("/short", function (req, res) {
...
+
   res.json(series);
 })
 .queryParam("start", startOffsetSchema)
 .queryParam("DBserver", clusterIdSchema)
 .summary("Short term history")
 .description("This function is used to get the statistics history.");
-
-
-router.get("/long", function (req, res) {
```
Comment (mchacki) [inline js/apps/system/_admin/aardvark/APP/statistics.js:478]: I am missing replacements for this function.
Is it not used by the UI anymore?

File: js/apps/system/_admin/aardvark/APP/statistics.js [hunk 3]
```diff
@@ -467,94 +464,10 @@ router.get("/short", function (req, res) {
...
-  res.json(series);
-})
-.queryParam("filter", joi.string().required())
-.queryParam("DBserver", clusterIdSchema)
-.summary("Long term history")
-.description("This function is used to get the aggregated statistics history.");
-
-
-router.get("/cluster", function (req, res) {
-  if (!cluster.isCoordinator()) {
```
Comment (mchacki) [inline js/apps/system/_admin/aardvark/APP/statistics.js:499]: I am missing replacements for this function.
Is it not used by the UI anymore?

**Review by mchacki — CHANGES_REQUESTED** [review]
I am missing two statistic methods now.
I am not unsure if they are obsolete or still used by UI.
If those two questions are sorted out I am very happy with this PR good work :+1:
Technically this is good

**Review by mchacki — APPROVED** [review]
Ok talked via Chat.
Sorted my comments out 👍 all good.

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/13558/

File: arangod/V8Server/V8DealerFeature.cpp [hunk 4]
```diff
@@ -365,6 +382,10 @@ void V8DealerFeature::prepare() {
 }
 
 void V8DealerFeature::start() {
+  LOG_DEVEL << "JS ENABLED!";
```
Comment (jsteemann) [inline arangod/V8Server/V8DealerFeature.cpp:385]:
```suggestion
```

**Comment by jsteemann** [comment]
Tests blue again

**Review by mchacki — APPROVED** [review]
LGTM

**Pull request merged**